src/stream/generator.ml

```diff
@@ -114,19 +114,19 @@ let remove_end g remove_len =
    * possibly cut some element in half, remove the rest.
    * The parsed elements are put back at the end
    * of the queue. *)
-  let rec cut l =
+  let rec cut ?(initial_ofs=0) l =
     let (c,ofs,len) = Queue.take g.buffers in
-      if len<l then begin
+      if len-initial_ofs<l then begin
         Queue.push (c,ofs,len) g.buffers ;
```

Comment:
I guess we could improve things by having buffers with view as for `Strings` here... (but this is not really related to the current PR).

Reply:
Agreed. The implementation of these buffers is not straight forward and could be enhanced indeed.